### PR TITLE
core: revert hlist for codecs

### LINKED ISSUE #110: revert hlist addition to codecs
I think addition of hlists to `Codecs` has been more pain than anything else, compile times are staggering once your protocol grows. Honestly I do not see what value it brings about. This is the commit that introduced hlists on `Codecs`:

https://github.com/oncue/remotely/commit/a087b00149d63af808136c148af330d27df4dc95

@timperrett @stew @runarorama WDYT?

If reverting that change is is not desirable, could we then at least find a better way that does not make scalac crap out once the protocol starts to grow?

### PR DESCRIPTION
**Pull request by ahjohannessen**
- revert adding hlists to `Codecs` and adjust
   dependant code / specs accordingly.

 - add convenience syntax for `Protocol` such
   that it is simpler to define (specify)
   signatures for protocols.

 - bump scala to 2.11.8

### CODE REVIEW

**Comment by ahjohannessen** [comment]
@timperrett WDYT?

File: core/src/main/scala/transport/netty/Transport.scala [hunk 1]
```diff
@@ -177,9 +177,9 @@ object Enframe extends ChannelOutboundHandlerAdapter {
     obj match {
       case Bits(bv) =>
         val byv = bv.toByteVector
```
Comment (ahjohannessen) [inline core/src/main/scala/transport/netty/Transport.scala]: @timperrett This is only here to see if PR builds successfully on Travis, I'll rebase on master when PR that @sbuzzard open is merged.

**Comment by timperrett** [comment]
@ahjohannessen so the PR build passed; that was what you were looking for, right? :-) I hope those changes i made to the travis.yml help too!

**Comment by ahjohannessen** [comment]
@timperrett I have rebased and think this is ready to be merged - I also bumped scala to 2.11.8 in travis and buil, WDYT? 

The caching addition is really great! :)

**Pull request merged**